cmake/tribits/common_tpls/FindTPLNetcdf.cmake

```diff
@@ -37,105 +37,130 @@
 # ************************************************************************
 # @HEADER
 
-#
-# First, set up the variables for the (backward-compatible) TriBITS way of
-# finding Netcdf.  These are used in case find_package(NetCDF ...) is not
-# called or does not find NetCDF.  Also, these variables need to be non-null
-# in order to trigger the right behavior in the function
-# tribits_tpl_find_include_dirs_and_libraries().
-#
+if (${CMAKE_VERSION} GREATER "3.13")
+     cmake_policy(SET CMP0074 NEW)
+endif()
 
-set(REQUIRED_HEADERS netcdf.h)
-set(REQUIRED_LIBS_NAMES netcdf)
+set(Netcdf_ALLOW_MODERN FALSE CACHE BOOL "Allow finding Netcdf as a modern CMake config file with exported targets (and only this way)")
 
-if (TPL_ENABLE_MPI)
-  set(REQUIRED_LIBS_NAMES ${REQUIRED_LIBS_NAMES} pnetcdf)
-endif()
+if (Netcdf_ALLOW_MODERN AND HDF5_FOUND_MODERN_CONFIG_FILE)
 
-#
-# Second, search for Netcdf components (if allowed) using the standard
-# find_package(NetCDF ...).
-#
-tribits_tpl_allow_pre_find_package(Netcdf  Netcdf_ALLOW_PREFIND)
-if (Netcdf_ALLOW_PREFIND)
-
-  message("-- Using find_package(Netcdf ...) ...")
-
-  set(CMAKE_MODULE_PATH
-    "${CMAKE_MODULE_PATH}"
-    "${CMAKE_CURRENT_LIST_DIR}/find_modules"
-    "${CMAKE_CURRENT_LIST_DIR}/utils"
-     )
-
-  find_package(NetCDF)
-
-  if (NetCDF_FOUND)
-    set(DOCSTR "List of semi-colon separated paths to look for the TPL Netcdf")
-    set(TPL_Netcdf_Enables_Netcdf4 ${NetCDF_NEEDS_HDF5} CACHE BOOL
-      "True if netcdf enables netcdf-4")
-    set(TPL_Netcdf_Enables_PNetcdf ${NetCDF_NEEDS_PNetCDF} CACHE BOOL
-      "True if netcdf enables pnetcdf")
-    set(TPL_Netcdf_PARALLEL ${NetCDF_PARALLEL} CACHE INTERNAL
-      "True if netcdf compiled with parallel enabled")
-    set(TPL_Netcdf_LIBRARY_DIRS ${_hdf5_LIBRARY_SEARCH_DIRS} CACHE PATH
-      "${DOCSTR} library files")
-    set(TPL_Netcdf_LIBRARIES ${NetCDF_LIBRARIES} CACHE PATH
-      "List of semi-colon separated library names (not 'lib' or extension).")
-    set(TPL_Netcdf_INCLUDE_DIRS ${NetCDF_INCLUDE_DIRS} CACHE PATH
-      "${DOCSTR} header files.")
+  set(minimum_modern_netCDF_version 4.7.4)
+  print_var(Netcdf_ALLOW_MODERN)
+  message("-- Using find_package(netCDF ${minimum_modern_netCDF_version} CONFIG) ...")
+  find_package(netCDF ${minimum_modern_netCDF_version} CONFIG)
```

Comment:
@ibaned, as per [above](https://github.com/trilinos/Trilinos/pull/11175/files#r1006168735), if `netCDFConfig.cmake` supports version ranges, I think this should be:

```
    find_package(netCDF ${minimum_modern_netCDF_version}... CONFIG)
```

no?

Reply:
As above, it was concluded that the current code already means that, but if it doesn't we will change it to mean that